### PR TITLE
Optimized database queries to address potential N+1 query issues

### PR DESCRIPTION
**Pull request by sagieramos**
- Solve the N+1 problem with "eager loading"
  - Before 
Screenshot from 2023-12-11 22-58-16
Screenshot from 2023-12-11 22-58-34
- After 
Screenshot from 2023-12-11 23-24-56
Screenshot from 2023-12-11 23-25-07

### CODE REVIEW

File: spec/views/user/index_html_spec.rb
Comment (LionRouge1) [inline spec/views/user/index_html_spec.rb:1]: - [x] Great job testing the user index page, all the testing are passing :clap: However it requested in the project requirement to test if the page have all the user's profile pictures. Therefore I suggest you add the test to assert the user's all the user's profile pictures on the user index page.
Screenshot from 2023-12-12 15-33-57

File: spec/views/post/index_html_spec.rb
Comment (LionRouge1) [inline spec/views/post/index_html_spec.rb:1]: - [ ] Nice work all the tests about the post index are passing. However you some tests cases are missing, Kindly add them in order to follow the project requirement:
  - Please write a test to assert the how many likes a post has.
  - Please write a test to assert the section for pagination if there are more posts than fit on the view
Screenshot from 2023-12-12 15-44-29

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @sagieramos & @Samuelwanza, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Solves `N+1` problem when fetching all posts and their comments :heavy_check_mark: 
- Use `Rspec` and `Capybara` gems for writing the tests :heavy_check_mark: 
- All the tests are passing :x:

## Required Changes ♻️
- [ ] Great job in general, all the tests for the views are passing :+1: However some tests for the models and requests are failing. Kindly check them and make sure all the tests in your app are passing.
Screenshot from 2023-12-12 15-56-10

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mckent05 — APPROVED** [review]
Hi @sagieramos, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Good job writing integration tests for your application 👍 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**